imod/mf6/utilities/imod5_converter.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+
+from imod.typing.grid import GridDataArray, full_like
+
+
+def convert_ibound_to_idomain(ibound: GridDataArray, thickness: GridDataArray):
```

Comment:
GridDataArray means it can be structured or unstructured. If we test only for structured, it may be better to specify xr.DataArray here (to not suggest it is more general than it really is)